controllers/infrastructure/byoadmission_controller.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	certv1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ByoAdmissionReconciler reconciles a ByoAdmission object
+type ByoAdmissionReconciler struct {
+	ClientSet clientset.Interface
+}
+
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=byohosts,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=byohosts/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=byohosts/finalizers,verbs=update
+//+kubebuilder:rbac:groups=certificates.k8s.io,resources=certificatesigningrequests,verbs=create;get;list;watch
+
+// Reconcile continuosuly checks for CSRs and approves them
+func (r *ByoAdmissionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var err error
+	logger := log.FromContext(ctx)
+	logger.Info("Reconcile request received", "object", req.NamespacedName)
+
+	// Fetch the CSR from the api-server
+	csr, err := r.ClientSet.CertificatesV1().CertificateSigningRequests().Get(ctx, req.NamespacedName.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Error(err, "CertificateSigningRequest not found, won't reconcile")
+			return reconcile.Result{}, errors.New("CertificateSigningRequest not found")
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Check if the CSR is already approved or denied
+	if checkCSRCondition(csr.Status.Conditions, certv1.CertificateApproved) {
```

Comment:
We maybe don't need a helper function here. CAPI provides https://pkg.go.dev/sigs.k8s.io/cluster-api@v1.1.3/util/conditions#Get

Sample usage - 
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/controllers/infrastructure/byomachine_controller_test.go#L160

Reply:
But the `CertificateSigningRequest` object doesn't satisfy the [Getter](https://pkg.go.dev/sigs.k8s.io/cluster-api@v1.1.3/util/conditions#Getter) interface.